clients/cobol-lsp-vscode-extension/src/provider/TarCopybookFileSystemProvider.ts

```diff
@@ -0,0 +1,200 @@
+import { Minimatch } from "minimatch";
+import * as vscode from "vscode";
+import { Memoize } from "../services/util/Memoize";
+import { TarUtil } from "../services/util/TarUtil";
+
+export class TarCopybookFileSystemProvider
+  implements vscode.FileSystemProvider
+{
+  public static readonly SCHEME = "cobol-ls-tar";
+  public tarCache = new Memoize(
+    async (tarFileUri: vscode.Uri) => {
+      return await TarUtil.readTarFile(tarFileUri);
+    },
+    undefined,
+    (tarFileUri: vscode.Uri) => {
+      const { tarfsPath, dialect } = this.getDetailsFromTarUri(tarFileUri);
+      return `${tarfsPath}$$${dialect}`;
+    },
+  );
+  private emitter = new vscode.EventEmitter<vscode.FileChangeEvent[]>();
+  readonly onDidChangeFile: vscode.Event<vscode.FileChangeEvent[]> =
+    this.emitter.event;
+
+  private watchers = new Set<WatchEntry>();
+
+  private fireChange(event: vscode.FileChangeEvent) {
+    this.emitter.fire([event]);
+
+    if (event.type === vscode.FileChangeType.Deleted) {
+      this.clearCache();
+    }
+  }
+  watch(
+    resource: vscode.Uri,
+    opts: { recursive: boolean; excludes: string[] },
+  ): vscode.Disposable {
+    const entry: WatchEntry = { resource, opts, disposed: false };
+    this.watchers.add(entry);
+
+    return new vscode.Disposable(() => {
+      entry.disposed = true;
+      this.watchers.delete(entry);
+    });
```

Comment:
We don't actually need to keep any kind of registrations for this.
```suggestion
    return { dispose(){} };
```

Reply:
removed at : c049c4198822676a3e08a4f57231b66f061c37d2